### PR TITLE
UX-619 convert ComboBox to typescript

### PR DESCRIPTION
**Pull request by logansparlin**
### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

**Review by bm-nhs — APPROVED** [review]

**Comment by bm-nhs** [comment]
😎lgtm

File: packages/matchbox/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  /**
+   * Maps to Downshift's refKey set in getRootProps. refKey must be set to "rootRef"
+   */
+  rootRef?: React.MutableRefObject<HTMLDivElement>;
+};
+
+function ComboBox(props: ComboBoxProps): JSX.Element {
+  const { children, style, rootRef, ...rest } = props;
+
+  function getChild(name?: string, passedProps?: { children?: ReactNode }) {
```
Comment (jonambas) [inline packages/matchbox/src/components/ComboBox/ComboBox.tsx:16]: Should we replace this with the other `getChild` helper?

File: packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx [hunk 2]
```diff
@@ -20,7 +19,21 @@ const StyledWrapper = styled('div')`
...
 `;
 
-function ComboBoxTextField(props) {
+export type ComboBoxTextFieldProps = {
+  delimiter?: string;
+  error?: string;
+  errorInLabel?: boolean;
+  helpText?: React.ReactNode;
+  id: string;
+  itemToString?: (any) => string;
```
Comment (jonambas) [inline packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx:28]: did you mean `(item: any)` ?

File: packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
```
Comment (jonambas) [inline packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx:2]: Can remove Proptypes

File: packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx (hunk 2 above)
Comment (jonambas) [reply packages/matchbox/src/components/ComboBox/ComboBoxTextField.tsx:28]: I guess you could type it to `string | { [k: string]: any }` but not sure if worth it

File: packages/matchbox/src/components/ComboBox/ComboBoxMenu.tsx [hunk 4]
```diff
@@ -1,14 +1,32 @@
 import React from 'react';
 import PropTypes from 'prop-types';
```
Comment (jonambas) [inline packages/matchbox/src/components/ComboBox/ComboBoxMenu.tsx:2]: Can remove PropTypes

**Review by jonambas — CHANGES_REQUESTED** [review]
Few comments

**Review by jonambas — APPROVED** [review]
Looks good

**Pull request merged**